### PR TITLE
Configure and enforce ktlint

### LINKED ISSUE #2: Kotlin linter to run on pre-commit

### PR DESCRIPTION
**Pull request by reicheltp**
Applies and enforces uniform Kotlin coding style via [ktlint][ktlint]. Adds a pre-comit hook.

Removes all `.idea` config as these depends on the developers os and hard pathes.

Adds `gradle initIdea` to easily generate a local userspecific `.idea` configuration.

Resolves #2 

[ktlint]: https://ktlint.github.io/

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Fri Sep 27 18:10:42 CEST 2019
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.3-all.zip
-distributionBase=GRADLE_USER_HOME
-distributionPath=wrapper/dists
-zipStorePath=wrapper/dists
-zipStoreBase=GRADLE_USER_HOME
+#Fri Sep 27 18:10:42 CEST 2019
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.4.1-all.zip
```
Comment (Sebkasanzew) [inline gradle/wrapper/gradle-wrapper.properties]: Why is there an `^M` at the line end?

File: .gitignore [hunk 2]
```diff
@@ -1,141 +1,114 @@
-
```
Comment (Sebkasanzew) [inline .gitignore]: I don't understand, what changed here

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (reicheltp) [reply gradle/wrapper/gradle-wrapper.properties]: 🤷‍♂

Comment (reicheltp) [reply gradle/wrapper/gradle-wrapper.properties]: Fixed

File: .gitignore (hunk 2 above)
Comment (reicheltp) [reply .gitignore]: Fixed

**Comment by reicheltp** [comment]
@Sebkasanzew line endings are fixed. Anything else todo?

**Comment by Sebkasanzew** [comment]
@reicheltp Ready to merge from my side

**Pull request merged**